pkg/query-service/app/logparsingpipeline/controller.go

```diff
@@ -0,0 +1,143 @@
+package logparsingpipeline
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/jmoiron/sqlx"
+	"go.signoz.io/signoz/pkg/query-service/agentConf"
+	"go.signoz.io/signoz/pkg/query-service/auth"
+	"go.signoz.io/signoz/pkg/query-service/model"
+	"go.uber.org/zap"
+)
+
+// Controller takes care of deployment cycle of log parsing pipelines.
+type LogParsingPipelineController struct {
+	Repo
+}
+
+func NewLogParsingPipelinesController(db *sqlx.DB, engine string) (*LogParsingPipelineController, error) {
+	repo := NewRepo(db)
+	err := repo.InitDB(engine)
+	return &LogParsingPipelineController{Repo: repo}, err
+}
+
+// PipelinesResponse is used to prepare http response for pipelines config related requests
+type PipelinesResponse struct {
+	*agentConf.ConfigVersion
+
+	Pipelines []model.Pipeline          `json:"pipelines"`
+	History   []agentConf.ConfigVersion `json:"history"`
+}
+
+// ApplyPipelines stores new or changed pipelines and initiates a new config update
+func (ic *LogParsingPipelineController) ApplyPipelines(ctx context.Context, postable []PostablePipeline) (*PipelinesResponse, *model.ApiError) {
+	// get user info from context
+	jwt, err := auth.ExtractJwtFromContext(ctx)
+	if err != nil {
+		return nil, model.InternalError(fmt.Errorf("failed to extract jwt from context %v", err))
+	}
+
+	claims, err := auth.ParseJWT(jwt)
+	if err != nil {
+		return nil, model.InternalError(fmt.Errorf("failed get claims from jwt %v", err))
+	}
+
+	userId := ""
+	if v, ok := claims["id"]; ok {
+		userId = v.(string)
+	}
+
+	var pipelines []model.Pipeline
+
+	// scan through postable pipelines, to select the existing pipelines or insert missing ones
+	for _, r := range postable {
+
+		// note: we process only new and changed pipelines here, deleted pipelines are not expected
+		// from client. if user deletes a pipelines, the client should not send that pipelines in the update.
+		// in effect, the new config version will not have that pipelines.
+
+		if r.Id == "" {
+			// looks like a new or changed pipeline, store it first
+			inserted, err := ic.insertPipeline(ctx, &r)
+			if err != nil || inserted == nil {
+				zap.S().Errorf("failed to insert edited pipeline", err)
+				return nil, model.BadRequestStr("failed to insert edited pipeline")
```

Comment:
So this will enter the pipelines one by one into the db ( the ones which are getting inserted are the new ones).

But until and unless they won't be used attached to a version in the agent_config_elements table these entries are of no use.